x/genutil/utils.go

```diff
@@ -65,7 +73,15 @@ func InitializeNodeValidatorFiles(config *cfg.Config) (nodeID string, valPubKey
 		return "", nil, err
 	}
 
-	tmValPubKey, err := privval.LoadOrGenFilePV(pvKeyFile, pvStateFile).GetPubKey()
+	var filePV *privval.FilePV
+	if !bip39.IsMnemonicValid(mnemonic) {
+		filePV = privval.LoadOrGenFilePV(pvKeyFile, pvStateFile)
+	} else {
```

Comment:
This reads a bit weird to me. Why can we not just check if the `mnemonic` is empty or not rather than invalid. In fact, if it's non-empty and invalid, this should return an error.

Reply:
@alexanderbez You're righ. I've fixed it as suggested and also added some tests